### ISSUE TITLE
Non-specific error message in publish modal

### ISSUE THREAD

**Issue by anikbrazeau** [cds-snc/platform-forms-client#7534]
## Description

Be more specific about how a user can recover from this error and what's missing to be able to continue : 

**Error** Please fill out all fields 

<img width="724" height="502" alt="Image" src="https://github.com/user-attachments/assets/b72b1417-32e2-4f16-98a5-8ed4a3a2147b" />
